### PR TITLE
Phase1 - Add support to read secure note

### PR DESCRIPTION
**Pull request by arotondo**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import org.gradle.api.initialization.resolve.RepositoriesMode
+
+plugins {
+    // Apply the foojay-resolver plugin to allow automatic download of JDKs
+    id 'org.gradle.toolchains.foojay-resolver-convention' version '0.7.0'
```
Comment (philippemin) [inline settings.gradle]: what is this for?

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.Value;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Value
```
Comment (philippemin) [inline src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: `@Value`?

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    private static final Duration DEFAULT_CONNECTION_TIMEOUT = Duration.ofSeconds(3);
+    private static final Duration DEFAULT_READ_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration DEFAULT_WRITE_TIMEOUT = Duration.ofSeconds(30);
+
+    String baseUrl;
+
+    @Getter(AccessLevel.PRIVATE)
+    @ToString.Exclude
+    OnePasswordConnectServerApiClient client;
```
Comment (philippemin) [inline src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: If you end up removing `@Value`, make sure these are private and immutable

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import lombok.ToString;
+import lombok.Value;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Value
+@NonNullByDefault
+public class OnePasswordSecrets {
```
Comment (philippemin) [inline src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: Rename this if you want to `OnePasswordConnector`

File: settings.gradle (hunk 1 above)
Comment (arotondo) [reply settings.gradle]: gone!

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java (hunk 2 above)
Comment (arotondo) [reply src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: Gone

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java (hunk 3 above)
Comment (arotondo) [reply src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: Done. I even removed `baseUrl` cause there is no need for it.

File: src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java (hunk 4 above)
Comment (arotondo) [reply src/main/java/com/ziro/onepassword/sdk/OnePasswordSecrets.java]: Done

**Comment by arotondo** [comment]
@philippemin this is ready for a second pass

**Review by philippemin — APPROVED** [review]
👍

**Pull request merged**